ckan/lib/helpers.py

```diff
@@ -1687,17 +1692,36 @@ def time_ago_from_timestamp(timestamp: int) -> str:
     return formatters.localised_nice_date(datetime_, show_date=False)
 
 
+def _object_display_name(obj_dict: dict[str, Any]) -> str:
+    return get_translated(obj_dict, 'title') or obj_dict['name']
+
+
 @core_helper
 def dataset_display_name(
         package_or_package_dict: Union[dict[str, Any], model.Package]) -> str:
+    if not isinstance(package_or_package_dict, dict):
+        package_or_package_dict = logic.get_action(u'package_show')(
+            {}, {u'id': package_or_package_dict.id})
+    return _object_display_name(package_or_package_dict)  # type: ignore
 
-    if isinstance(package_or_package_dict, dict):
-        return get_translated(package_or_package_dict, 'title') or \
-            package_or_package_dict['name']
-    else:
-        # FIXME: we probably shouldn't use the same functions for
-        # package dicts and real package objects
-        return package_or_package_dict.title or package_or_package_dict.name
+
+@core_helper
+def group_display_name(
+        group_or_group_dict: Union[dict[str, Any], model.Group]) -> str:
+    if not isinstance(group_or_group_dict, dict):
+        group_type = group_or_group_dict.type
+        group_id = group_or_group_dict.id
+    elif 'display_name' not in group_or_group_dict:
+        group_type = group_or_group_dict[u'type']
+        group_id = group_or_group_dict[u'id']
```

Comment:
why do we need to call *_show if display_name is missing? We're not using it now, right?

Reply:
@wardi I did not fully remove the display_name from the groups, as it was used in a lot of places still, and seems like packages also has some display name thing going on with them.

If I am remembering correctly, these conditions were for the facets, in which the passed dict of the facet item does not contain the organization object or anything.

Should I do a bunch more stuff to just fully remove the display_name from groups, and then figure out the facet stuff later?

Reply:
I recall now that I think the condition is for the breadcrumb, not the facet.

Reply:
Removed all of these conditions, as we should really already have the dicts and `_translated` fields at this point. So we should not be doing that in the helpers here. This helper is only really returning the `get_translated` title OR name.